sci-misc/boinc/files/boinc.init

```diff
@@ -9,46 +9,91 @@ depend() {
 	use dns net ntp-client ntpd
 }
 
-
 create_work_directory() {
-	if [[ ! -d "${RUNTIMEDIR}" ]]; then
+	local sslcrt="/etc/ssl/certs/ca-certificates.crt"
+
+	if [ ! -d "${RUNTIMEDIR}" ]; then
 		einfo "Directory ${RUNTIMEDIR} does not exist, creating now."
 		mkdir -p "${RUNTIMEDIR}"
-		if [[ ! -d "${RUNTIMEDIR}" ]]; then
+		if [ $? -ne 0 ]; then
 			eeror "Directory ${RUNTIMEDIR} could not be created!"
 			return 1
 		fi
 
 		# ensure proper ownership
 		chown "${USER}:${GROUP}" "${RUNTIMEDIR}"
+		if [ $? -ne 0 ]; then
+			eeror "Changing ownership of '${RUNTIMEDIR}' to '${USER}:${GROUP}' failed!"
+			return 1
+		fi
 	fi
 
-	if [[ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]]; then
-		ln -s /etc/ssl/certs/ca-certificates.crt "${RUNTIMEDIR}"/ca-bundle.crt
+	if [ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]; then
+
+		if [ -f "${sslcrt}" ]; then
+			ln -s "${sslcrt}" "${RUNTIMEDIR}"/ca-bundle.crt
+			if [ $? -ne 0 ]; then
+				eeror "Symlinking '${sslcrt}' failed!"
+				return 1
+			fi
+		else
+			eerror "'${sslcrt}' does not exist!"
+			return 1
+		fi
 	fi
 
 	return 0
 }
 
+fix_lib_symlinks() {
+	local src="$1"
+	local tgt="$2"
 
-cuda_check() {
-	local libtarget="${RUNTIMEDIR}/libcudart.so"
-	local libsource="$(ls -t /opt/cuda/lib*/libcudart.so 2>/dev/null | head -n 1)"
+	# If the source does not exist, we can not do anything
+	if [ ! -f "${src}" ] ; then
+		return 1
+	fi
 
-	# Remove a broken symlink
-	if [[ -h "${libtarget}" ]] \
-	&& [[ "${libsource}" != "$(readlink "${libtarget}")" ]]; then
-		rm -f "${libtarget}"
+	# Check whether the symlink is already there and in order
+	if [ -L "${tgt}" ] ; then
+		if [ -f "$tgt" ] ; then
```

Comment:
You are using `${foo}` and `$foo` inconsistently. Please settle on one, preferably the first one.